erigon-lib/commitment/hex_patricia_hashed.go

```diff
@@ -1659,16 +1602,19 @@ func (hph *HexPatriciaHashed) fold() (err error) {
 
 		if hph.branchBefore[row] { // encode Delete if prefix existed before
 			//fmt.Printf("delete existed row %d prefix %x\n", row, updateKey)
-			_, err := hph.branchEncoder.CollectUpdate(hph.ctx, updateKey, 0, hph.touchMap[row], 0, RetrieveCellNoop)
+			del, err := hph.branchEncoder.EncodeDelete(hph.touchMap[row])
 			if err != nil {
 				return fmt.Errorf("failed to encode leaf node update: %w", err)
 			}
+			if err = hph.ctx.PutBranch(common.Copy(updateKey), common.Copy(del), nil, 0); err != nil {
```

Comment:
Inside `ctx.PutBranch`:
- `AddPrevValue` - does copy inside
- `DomainUpdate` - does copy inside
in this case: `ctx.PutBranch` also can do copy inside?